src/js/providers/utils/play-promise.ts

```diff
@@ -1,22 +1,30 @@
-export default function createPlayPromise(video) {
-    return new Promise(function(resolve, reject) {
+type Resolve = () => void;
+
+type Reject = (reason: PlayPromiseError) => void;
+
+type PlayPromiseError = Error & {
```

Comment:
aw man we could use `PlayerError` if we had a `sourceError` prop here. oh well, more potential refactoring that TS is showcasing. good for now